File: temp/agent/lib/src/models/enrollment.dart
class EnrollmentFormField {
  final String label;
  final String type;
  final bool required;
  final List<String>? options;

  EnrollmentFormField({
    required this.label,
    required this.type,
    required this.required,
    this.options,
  });

  factory EnrollmentFormField.fromJson(Map<String, dynamic> json) {
    return EnrollmentFormField(
      label: json['label'],
      type: json['type'],
      required: json['required'] ?? false,
      options: json['options'] != null ? List<String>.from(json['options']) : null,
    );
  }
}

class EnrollmentConfig {
  final String id;
  final String courseId;
  final List<EnrollmentFormField> fields;
  final bool requireImages;
  final int imageCount;
  final String? imageInstructions;

  EnrollmentConfig({
    required this.id,
    required this.courseId,
    required this.fields,
    required this.requireImages,
    required this.imageCount,
    this.imageInstructions,
  });

  factory EnrollmentConfig.fromJson(Map<String, dynamic> json) {
    return EnrollmentConfig(
      id: json['id'],
      courseId: json['course_id'],
      fields: (json['fields'] as List)
          .map((e) => EnrollmentFormField.fromJson(e))
          .toList(),
      requireImages: json['require_images'] ?? false,
      imageCount: json['image_count'] ?? 1,
      imageInstructions: json['image_instructions'],
    );
  }
}

class EnrollmentRequest {
  final String id;
  final String userId;
  final String courseId;
  final String status;
  final Map<String, dynamic> formData;
  final String? adminComment;
  final DateTime createdAt;

  EnrollmentRequest({
    required this.id,
    required this.userId,
    required this.courseId,
    required this.status,
    required this.formData,
    this.adminComment,
    required this.createdAt,
  });

  factory EnrollmentRequest.fromJson(Map<String, dynamic> json) {
    return EnrollmentRequest(
      id: json['id'],
      userId: json['user_id'],
      courseId: json['course_id'],
      status: json['status'],
      formData: Map<String, dynamic>.from(json['form_data'] ?? {}),
      adminComment: json['admin_comment'],
      createdAt: DateTime.parse(json['created_at']),
    );
  }
}

File: temp/main-server/app/schemas/enrollment.py
from typing import Optional, List, Dict, Any
from pydantic import BaseModel
from datetime import datetime


class EnrollmentFormField(BaseModel):
    label: str
    type: str  # text, number, select, date
    required: bool = False
    options: Optional[List[str]] = None


class EnrollmentFormConfigBase(BaseModel):
    fields: List[EnrollmentFormField] = []
    require_images: bool = False
    image_count: int = 1
    image_instructions: Optional[str] = None


class EnrollmentFormConfigCreate(EnrollmentFormConfigBase):
    pass


class EnrollmentFormConfigResponse(EnrollmentFormConfigBase):
    id: str
    course_id: str
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class EnrollmentRequestImageResponse(BaseModel):
    id: str
    url: str
    created_at: datetime

    class Config:
        from_attributes = True


class EnrollmentRequestBase(BaseModel):
    course_id: str
    form_data: Dict[str, Any]


class EnrollmentRequestCreate(EnrollmentRequestBase):
    # Images will be handled via multipart/form-data or separate URLs
    image_urls: List[str] = []


class EnrollmentRequestResponse(BaseModel):
    id: str
    user_id: str
    course_id: str
    status: str
    form_data: Dict[str, Any]
    admin_comment: Optional[str] = None
    images: List[EnrollmentRequestImageResponse] = []
    created_at: datetime
    updated_at: datetime
    processed_at: Optional[datetime] = None
    processed_by: Optional[str] = None
    
    # Extra info for admin
    user_email: Optional[str] = None
    course_title: Optional[str] = None

    class Config:
        from_attributes = True
